scripts/src/javatests/org/oppia/android/scripts/regex/RegexPatternValidationCheckTest.kt

```diff
@@ -1687,6 +1707,434 @@ class RegexPatternValidationCheckTest {
     )
   }
 
+  @Test
+  fun testFileContent_doesNotHaveColorSuffixOrSnakeCasing_fileContentIsNotCorrect() {
+    val prohibitedContent =
+      """
+      <color name="color_palette_background_color">@color/oppia_light_yellow</color>
+      <color name="color_palette_darkBackgroundColor">@color/mid_grey_30</color>
+      <color name="color_palette_description_text_color">@color/accessible_light_grey</color>
+      <color name="color_palette_text_input_background">@color/white</color>
+      <color name="color_palette_DarkText">@color/black_87</color>
+      <color name="color_palette_error_color">@color/oppia_red</color>
+      <color name="color_palette_CONTAINER_BACKGROUND_COLOR">@color/white</color>
+      <color name="color_palette_toolbar">@color/oppia_green</color>
+      """.trimIndent()
+    tempFolder.newFolder("testfiles", "app", "src", "main", "res", "values")
+    val stringFilePath = "app/src/main/res/values/color_palette.xml"
+    tempFolder.newFile("testfiles/$stringFilePath").writeText(prohibitedContent)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    // Verify that all patterns are properly detected & prohibited.
+    assertThat(exception).hasMessageThat().contains(REGEX_CHECK_FAILED_OUTPUT_INDICATOR)
+    assertThat(outContent.toString().trim())
+      .isEqualTo(
+        """
+        $stringFilePath:2: $doesNotHaveColorSuffixOrSnakeCasing
+        $stringFilePath:4: $doesNotHaveColorSuffixOrSnakeCasing
+        $stringFilePath:5: $doesNotHaveColorSuffixOrSnakeCasing
+        $stringFilePath:7: $doesNotHaveColorSuffixOrSnakeCasing
+        $stringFilePath:8: $doesNotHaveColorSuffixOrSnakeCasing
+        $wikiReferenceNote
+        """.trimIndent()
+      )
+  }
+
+  @Test
+  fun testFileContent_hasColorSuffixAndSnakeCasing_fileContentIsCorrect() {
+    val prohibitedContent =
+      """
+        <color name="color_palette_toolbar_color">@color/oppia_green</color>
+        <color name="color_palette_status_bar_color">@color/dark_green</color>
+        <color name="color_palette_action_bar_color">@color/oppia_green</color>
+      """.trimIndent()
+    tempFolder.newFolder("testfiles", "app", "src", "main", "res", "values")
+    val stringFilePath = "app/src/main/res/values/color_palette.xml"
```

Comment:
I suggest updating the new test names for tests that are verifying a specific file to be specific to that file, e.g. ``testFileContent_colorPalette_...``.

Reply:
I did that for every but one test as renaming that function exceeds the 100 characters limit. Can you suggest something for it?

Reply:
Ah, are you referring to ``testFileContent_doesNotReferenceColorFromColorPalette_fileContentIsNotCorrect``?

Perhaps:

``testFileContent_componentColors_includesNonPaletteColorReferences_fileContentIsNotCorrect``

Reply:
Done.